File: my-folder/1392-find-the-difference-of-two-arrays/solution.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {
    // let answer1 = [];
    // let answer2 = [];
    // let answer = [];

    // nums1 = [...new Set(nums1)];
    // nums2 = [...new Set(nums2)];
    
    // for (let i = 0; i < nums1.length; i++) {
    //     if (!nums2.includes(nums1[i])) {
    //         answer1.push(nums1[i]);
    //     }
    // }
    // for (let i = 0; i < nums2.length; i++) {
    //     if (!nums1.includes(nums2[i])) {
    //         answer2.push(nums2[i]);
    //     }
    // }

    // answer.push(answer1);
    // answer.push(answer2);
    // return  answer;   

    const set1 = new Set(nums1);
    const set2 = new Set(nums2);

    const answer1 = [...set1].filter(num => !set2.has(num));
    const answer2 = [...set2].filter(num => !set1.has(num));

    return [answer1, answer2];
};

File: my-folder/1392-find-the-difference-of-two-arrays/solution.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        # set1 = set(nums1)
        # set2 = set(nums2)

        # set1Diff = set1.difference(set2)
        # set2Diff = set2.difference(set1)

        # return [list(set1Diff), list(set2Diff)]
        
        return [list(set(nums1) - set(nums2)), list(set(nums2) - set(nums1))]
